pkg/authorization/cache/clusterpolicy.go

```diff
@@ -24,7 +24,7 @@ type readOnlyClusterPolicyCache struct {
 	keyFunc cache.KeyFunc
 }
 
-func NewReadOnlyClusterPolicyCache(registry clusterpolicyregistry.WatchingRegistry) readOnlyClusterPolicyCache {
+func NewReadOnlyClusterPolicyCache(registry clusterpolicyregistry.WatchingRegistry) *readOnlyClusterPolicyCache {
```

Comment:
> that mutex?

Looks like the mutex inside of the reflector inside of the cache.  The correct fix depends on copy semantics, but I think that this API is more correct based on the rest of the code.